tools/clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -1047,8 +1047,15 @@ bool CGDebugInfo::TryCollectHLSLRecordElements(const RecordType *Ty,
     unsigned VecSize = hlsl::GetHLSLVecSize(QualTy);
     unsigned ElemSizeInBits = CGM.getContext().getTypeSize(ElemQualTy);
     unsigned CurrentAlignedOffset = 0;
+    std::string FieldNameBuf;
```

Comment:
This way we don't have dynamic allocations until we get pretty big.
```suggestion
    SmallString<16> FieldNameBuf;
```

Reply:
I agree with using `SmallString`, though `16` seems even larger than necessary.  I think `"c1023"` is the largest string we can get here, so I think `8` should be sufficient, don't you?

Reply:
Sure. It’s also only 8 bytes of stack space but we probably don’t need more than 5 bytes (8 is the practical minimum since the structure has 8-byte alignment requirement).

Reply:
Updated to use `SmallString<8>`